tests/console/zfs.pm

```diff
@@ -0,0 +1,226 @@
+    assert_script_run('zpool list | grep tank');
+    assert_script_run('zpool create dozer /var/tmp/dozer.img -o ashift=12');
+    assert_script_run('zpool list | grep dozer');
+    assert_script_run('cd /tank/; ls');
+    # Put data in tank
+    assert_script_run('curl -v -o /tank/Big_Buck_Bunny_8_seconds_bird_clip.ogv ' . data_url('Big_Buck_Bunny_8_seconds_bird_clip.ogv'));
+    assert_script_run('md5sum Big_Buck_Bunny_8_seconds_bird_clip.ogv > /var/tmp/Big_Buck_Bunny_8_seconds_bird_clip.ogv.md5sum');
+    assert_script_run('md5sum -c /var/tmp/Big_Buck_Bunny_8_seconds_bird_clip.ogv.md5sum');
+    scrub('tank');
+    assert_script_run('zpool status tank | grep state | grep ONLINE');
+    script_run('zpool status tank');
+
+    ## Test raidz capability by corruping a disk
+    corrupt_disk('/var/tmp/tank_a.img');
+    assert_script_run('md5sum -c /var/tmp/Big_Buck_Bunny_8_seconds_bird_clip.ogv.md5sum');
+    scrub('tank');
+    assert_script_run("zpool status tank | grep scan | grep 'scrub repaired'");
+    assert_script_run('md5sum -c /var/tmp/Big_Buck_Bunny_8_seconds_bird_clip.ogv.md5sum');
+    script_run('zpool status tank');
+    assert_script_run('zpool status tank | grep "corrupted data"');
+    # Replace disk
+    assert_script_run('zpool offline tank /var/tmp/tank_a.img');
+    clear_disk('/var/tmp/tank_a.img');
+    assert_script_run('zpool replace tank /var/tmp/tank_a.img');
+    scrub('tank');
+    assert_script_run('zpool status tank | grep state | grep ONLINE');
+    # Display status for debugging purposes
+    script_run('zpool status tank');
+
+    ## Test Snapshots
+    assert_script_run('zfs snapshot tank@initial');
+    assert_script_run('zfs list -t snapshot | grep "tank@initial"');
+    assert_script_run('curl -v -o /tank/test_unzip.zip ' . data_url('console/test_unzip.zip'));
+    assert_script_run('rm /tank/Big_Buck_Bunny_8_seconds_bird_clip.ogv');
+    # Create second snapshot with other data
+    assert_script_run('zfs snapshot tank@second');
+    assert_script_run('zfs snapshot tank@third');
+    assert_script_run('zfs list -t snapshot | grep "tank@second"');
+    assert_script_run('zfs list -t snapshot | grep "tank@third"');
+    # Mount snapshots
+    assert_script_run('mkdir -p /mnt/tank/{initial,second}');
+    assert_script_run('mount -o ro -t zfs tank@initial /mnt/tank/initial');
+    assert_script_run('zfs clone -o mountpoint=/mnt/tank/second tank@second tank/2nd_second');
+    assert_script_run('stat /mnt/tank/initial/Big_Buck_Bunny_8_seconds_bird_clip.ogv');
+    assert_script_run('cd /mnt/tank/initial/; md5sum -c /var/tmp/Big_Buck_Bunny_8_seconds_bird_clip.ogv.md5sum');
+    assert_script_run('! stat /mnt/tank/initial/test_unzip.zip');
+    assert_script_run('! stat /mnt/tank/second/Big_Buck_Bunny_8_seconds_bird_clip.ogv');
+    assert_script_run('stat /mnt/tank/second/test_unzip.zip');
+    assert_script_run('cd');
+    assert_script_run('umount /mnt/tank/{initial,second}');
```

Comment:
clone named second should be unmout by zfs it self:
`zfs unmount tank/2nd_second`

Reply:
`zfs umount` seems to not work with snapshots: http://phoenix-openqa.qam.suse.de/tests/2721#step/zfs/180

However, I added a `zfs umount` check at the end of the test